plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/RenameElementRefactoringProcessor.java

```diff
@@ -58,10 +66,39 @@ public RefactoringStatus checkFinalConditions(final IProgressMonitor pm, final C
 	@Override
 	public Change createChange(final IProgressMonitor pm) throws CoreException, OperationCanceledException {
 		final CompositeChange change = new CompositeChange(getProcessorName());
-		change.add(new RenameElementChange(elementURI.lastSegment(), elementURI, newName));
+		change.add(new RenameElementChange("Rename Pin in type: " + elementURI.lastSegment(), elementURI, newName));
+		createChildChanges(change);
 		return change;
 	}
 
+	private void createChildChanges(final CompositeChange change) {
+		final TypeEntry typeEntry = TypeLibraryManager.INSTANCE.getTypeEntryForURI(elementURI);
+		final BlockTypeInstanceSearch search = new BlockTypeInstanceSearch(typeEntry);
+		final List<? extends EObject> result = search.performSearch();
+		final var eChild = getChildByURI(typeEntry.getType(), elementURI);
+		String oldName = "";
+		if (eChild instanceof final IInterfaceElement varDecl) {
+			oldName = varDecl.getName();
+		}
+
+		for (final EObject eObject : result) {
+			change.add(new ReconnectPinChange(EcoreUtil.getURI(eObject), FBNetworkElement.class, newName, oldName));
+		}
+
+	}
+
+	public static EObject getChildByURI(final EObject parent, final URI uri) {
+		final EObject[] found = { null };
+		parent.eAllContents().forEachRemaining(child -> {
+			final String uriFragment = child.eResource().getURIFragment(child);
+			if (uriFragment.equals(uri.fragment())) {
+				found[0] = child;
+			}
+
+		});
+		return found[0];
+	}
```

Comment:
Please just use `parent.eResource().getEObject(uri.fragment())` instead of searching through the entire contents.